### PR TITLE
asserts,o/devicestate: support model specified alternative serial-authority

### PR DESCRIPTION
**Pull request by pedronis**
Support a new header in models to list alternative authorities that can sign model serials.

Accept such serials on registration.

Once the store has implemented this we can add a spread test.

### CODE REVIEW

**Comment by pedronis** [comment]
This has some real spread issues.

File: overlord/devicestate/devicestatetest/devicesvc.go [hunk 1]
```diff
@@ -174,7 +178,11 @@ func MockDeviceService(c *C, bhv *DeviceServiceBehavior) *httptest.Server {
 				"device-key-sha3-384": serialReq.SignKeyID(),
 				"timestamp":           time.Now().Format(time.RFC3339),
 			}, serialReq.Body())
-			c.Assert(err, IsNil)
+			c.Check(err, IsNil)
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (pedronis) [reply overlord/devicestate/devicestatetest/devicesvc.go]: Assert outside of the test goroutine, especially inside the http server code I think, produce hard to understand failures, so I use a Check, but then we need to give back an error to the client not do silly things

[the remark this answers is not in the document]
Comment (stolowski) [reply overlord/devicestate/devicestatetest/devicesvc.go]: Ah, I see, mind adding a comment in the code?

[the remark this answers is not in the document]
Comment (pedronis) [reply overlord/devicestate/devicestatetest/devicesvc.go]: to be clear, I think one issue is that http server does recover in some cases. But I don't remember the details. I would have to play with these issues more in isolation

**Pull request merged**